pom.xml

```diff
@@ -71,15 +66,15 @@
     <repository>
       <id>spigot-repo</id>
       <name>Spigot Maven Repository</name>
-      <url>https://hub.spigotmc.org/nexus/content/groups/public</url>
+      <url>https://hub.spigotmc.org/nexus/content/repositories/snapshots/</url>

```

Comment:
ここなんですが、なぜpublicリポジトリからsnapshotリポジトリへ変更されているのですか？
bukkit 1.15.2 の未リリースの変更内容が参照されてしまうので、あまり望ましくないと思います。

Reply:
公式の推奨するレポジトリを使用するほうが良いと判断しました。

BuildToolsで生成される、1.15.2のcraftbukkitのpom.xmlにこんな文言があります。
```xml
    <repositories>
        <!--
            If you are a plugin developer, please use https://hub.spigotmc.org/nexus/content/repositories/snapshots/
            as your repository URL. This will ensure only Bukkit / Spigot-API are pulled from our Maven repository.

            Please see https://www.spigotmc.org/go/maven for more information.
        -->
        <repository>
            <id>spigotmc-public</id>
            <url>https://hub.spigotmc.org/nexus/content/groups/public/</url>
        </repository>
    </repositories>
```

Reply:
> bukkit 1.15.2 の未リリースの変更内容が参照されてしまうので、あまり望ましくないと思います。

私には実際に未リリースの変更内容が参照されてしまうのかどうか、判断が付きませんでした。
解説しているサイトを検索しても、どちらを使っているかは微妙です。
Spigotの解説ページではsnapshotのほうを使っているようでしたが、その他のページではpublicの方を使ってることもありました。

APIは使うべきスコープがprovidedであり、バージョンが変わらないうちにメソッドが消えたり増えたりすることは無いと思うので、どちらを使うかは好みなのかもしれません。

Reply:
公式の推奨するレポジトリがあるのは知りませんでした。おっしゃる通り、snapshotsリポジトリを参照するのが正しいようです。次の機会に修正しておきます。

そもそも、私のコメントの「snapshotsリポジトリよりpublicリポジトリを使うべき」は誤りでした。「snapshotsリポジトリよりreleasesリポジトリを使うべき」が正しいですね。私の認識違いでおかしなコメントになってしまっていました。お詫びいたします。
Mavenリポジトリの違いについては、
https://terasolunaorg.github.io/guideline/5.0.1.RELEASE/ja/Appendix/Nexus.html
のサイトの7.4.2.章の最後辺りに書いてありますね。

CraftBukkitのリリースなんですが、昔は（1.4とか1.5とかの頃は）SNAPSHOTバージョンでの開発がしばらく続いて、ある程度品質が落ち着いたらリリースバージョンが発行されていたんですよね。R0.1-SNAPSHOT、R0.2-SNAPSHOT、… の後に、R1.0とかR2.0とかがリリースされていました。
今のバージョンではもう、SNAPSHOTしかなくてリリースバージョンがないですよね。それで、「snapshotsリポジトリを直接参照せよ」というガイドラインなのでしょう。